src/include/concepts.h

```diff
@@ -194,6 +200,14 @@ concept contiguous_partitioned_feature_vector_array =
       { d.ids() };
     };
 
+// ----------------------------------------------------------------------------
+// matrix_with_ids concept
+// ----------------------------------------------------------------------------
+template <class D>
+concept matrix_with_ids = requires(D d) {
+  { d.ids() };
```

Comment:
> But a higher level question -- what do we need this concept for?
We need this for this code:
```
  template <feature_vector_array T>
  explicit FeatureVectorArray(T&& obj)
      : vector_array(
            std::make_unique<vector_array_impl<T>>(std::forward<T>(obj))) {
    feature_type_ = tiledb::impl::type_to_tiledb<
        typename std::remove_cvref_t<T>::value_type>::tiledb_type;
    feature_size_ = datatype_to_size(feature_type_);

    if constexpr (matrix_with_ids<std::remove_cvref_t<decltype(obj)>>) {
      ids_type_ = tiledb::impl::type_to_tiledb<
          typename std::remove_cvref_t<T>::ids_type>::tiledb_type;
      ids_size_ = datatype_to_size(ids_type_);
    }
  }
```
What is your suggestion instead of this? I updated to `feature_vector_array_with_ids`, do you prefer that? I did have to make a change to get it to build, without the update to be:
```
template <class D>
concept feature_vector_array_with_ids =
    feature_vector_array<D> && requires(D d) {
      { d.ids() };
    };
```
I would get:
```
if constexpr (feature_vector_array_with_ids<std::remove_cvref_t<decltype(obj)>>) {
      ids_type_ = tiledb::impl::type_to_tiledb<
          typename std::remove_cvref_t<T>::ids_type>::tiledb_type;
      ids_size_ = datatype_to_size(ids_type_);
    }
```
I get:
```
/Users/parismorgan/repo/TileDB-Vector-Search-2/src/include/api/feature_vector_array.h:76:44: error: no type named 'ids_type' in 'Matrix<int, Kokkos::layout_left>'
          typename std::remove_cvref_t<T>::ids_type>::tiledb_type;
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
```